File: question/neetcode150/binary_search/find-minimum-in-rotated-sorted-array.py
from typing import List

class Solution:
    def findMin(self, a: List[int]) -> int:
        f = a[0]

        i = 0
        n = len(a)
        while n:
            while i+n < len(a) and a[i+n] > f:
                i+=n
            n//=2

        return a[i+1] if i+1 != len(a) else f

File: question/neetcode150/binary_search/find-minimum-in-rotated-sorted-array.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int findMin(vector<int>& a) {
        int n = a.size();
        int f = a[0];

        int k = 0;

        for(int i=n;i;i/=2){
            while(k+i < n and a[k+i]>f){
                k+=i;
            }
        }

        if(n==k+1){
            return f;
        }
        return a[k+1];
    }
};